client/src/main.tsx

```diff
@@ -3,39 +3,56 @@ import ReactDOM from "react-dom/client";
 
 import { Login, Register, Loading, NotFound, EditProfile, Search } from "./components/pages";
 
-import { createBrowserRouter, RouterProvider, redirect, Outlet } from "react-router-dom";
+import { createBrowserRouter, RouterProvider, defer } from "react-router-dom";
 
 import { ChakraProvider } from "@chakra-ui/react";
 import { userProfileSchema } from "./schemas.ts";
+import { useProfileStore } from "./store.ts";
+import { getToken } from "./token.ts";
 
-const router = createBrowserRouter([
-  {
-    path: "/",
-    loader: async () => {
-      const token = localStorage.getItem("token") ?? null;
+const implicitLogin = async () => {
+  try {
+    const profile = useProfileStore.getState();
+    if (profile.displayName !== "") return null;
+
+    const token = getToken();
+    if (!token) throw new Error("Token doesn't exist");
 
-      try {
-        if (!token) throw new Error("No such token");
+    const response = await fetch("localhost:8080/implicit_login", {
```

Comment:
Instead of hard-coding the host name, can we use the environment variable here? If we ever were to deploy this app `localhost` would be incorrect

https://vitejs.dev/guide/env-and-mode.html#env-files

I think one of our pages already uses the environment variable, so you can use that as a reference